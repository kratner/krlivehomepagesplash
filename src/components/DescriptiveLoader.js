import React from "react";

export default props => {
  const injectStyle = style => {
    const styleElement = document.createElement("style");
    let styleSheet = null;

    document.head.appendChild(styleElement);

    styleSheet = styleElement.sheet;

    styleSheet.insertRule(style, styleSheet.cssRules.length);
  };

  const keyframesStyle = `
    @-webkit-keyframes pulse {
        0%   { opacity: 0; }
        50%  { opacity: 1; }
    }
    `;
  injectStyle(keyframesStyle);
  const animation = {
    WebkitAnimation: "pulse 1s linear infinite"
  };
  const messageStyle = {
    color: props.messageColor
  };
  return (
    <div style={messageStyle}>
      <span>{props.message}</span>
      <span style={animation}>{props.animatedMessage}</span>
    </div>
  );
};
