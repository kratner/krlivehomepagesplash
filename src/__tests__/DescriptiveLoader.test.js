import React from "react";
import { shallow } from "enzyme";
import DescriptiveLoader from "../components/DescriptiveLoader";

describe("DescriptiveLoader", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<DescriptiveLoader message="Loading Message..." />);
    expect(wrapper).toMatchSnapshot();
  });
});
