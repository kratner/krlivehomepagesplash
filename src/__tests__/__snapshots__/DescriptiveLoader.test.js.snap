// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptiveLoader renders correctly 1`] = `ShallowWrapper {}`;
